master/master.cfg

```diff
@@ -100,31 +100,34 @@ HALIDE_BRANCHES = {HALIDE_MAIN: VersionedBranch(ref='master', version=Version(12
+WorkerConfig = namedtuple('WorkerConfig', ['max_builds', 'j', 'arch', 'bits', 'os'])
+
+# For machines with max_builds=1, using nproc+2 cores for building is the conventional choise
+# (and what ninja defaults to). Oddly, "ninja -j 0" means "use as many threads as you like" which
+# is definitely not what we want.
+_NPROC_PLUS_2 = '$(($(nproc)+2))'
 
 _WORKERS = [
-    ('linux-worker-1', WorkerConfig(max_builds=4, arch='x86', bits=[32, 64], os='linux')),
-    # Disable linuxbot 2 and 3 for now, since they are *much* slower than 1 and 4
-    # ('linux-worker-2', WorkerConfig(max_builds=2, arch='x86', bits=[32, 64], os='linux')),
-    # ('linux-worker-3', WorkerConfig(max_builds=2, arch='x86', bits=[32, 64], os='linux')),
-    ('linux-worker-4', WorkerConfig(max_builds=4, arch='x86', bits=[32, 64], os='linux')),
-    ('mac-worker-1', WorkerConfig(max_builds=2, arch='x86', bits=[64], os='osx')),
-    ('mac-arm-worker-1', WorkerConfig(max_builds=2, arch='arm', bits=[64], os='osx')),
-    ('arm32-linux-worker-1', WorkerConfig(max_builds=1, arch='arm', bits=[32], os='linux')),
-    ('arm32-linux-worker-2', WorkerConfig(max_builds=1, arch='arm', bits=[32], os='linux')),
-    ('arm64-linux-worker-1', WorkerConfig(max_builds=1, arch='arm', bits=[64], os='linux')),
-    ('arm64-linux-worker-2', WorkerConfig(max_builds=1, arch='arm', bits=[64], os='linux')),
-    # TODO: this bot is configured to handle both arm64 and arm32 builds. For now, just running arm32.
-    # ('rpi4-linux-worker-1', WorkerConfig(max_builds=1, arch='arm', bits=[32, 64], os='linux')),
-    ('rpi4-linux-worker-1', WorkerConfig(max_builds=1, arch='arm', bits=[32], os='linux')),
-    ('win-worker-1', WorkerConfig(max_builds=1, arch='x86', bits=[32, 64], os='windows')),
-    ('win-worker-2', WorkerConfig(max_builds=1, arch='x86', bits=[32, 64], os='windows')),
+    ('linux-worker-1', WorkerConfig(max_builds=4, j=8, arch='x86', bits=[32, 64], os='linux')),
+    ('linux-worker-4', WorkerConfig(max_builds=4, j=8, arch='x86', bits=[32, 64], os='linux')),
+    ('mac-worker-1', WorkerConfig(max_builds=2, j=8, arch='x86', bits=[64], os='osx')),
+    ('mac-arm-worker-1', WorkerConfig(max_builds=2, j=8, arch='arm', bits=[64], os='osx')),
+    ('arm32-linux-worker-1', WorkerConfig(max_builds=1, j=_NPROC_PLUS_2, arch='arm', bits=[32], os='linux')),
+    ('arm32-linux-worker-2', WorkerConfig(max_builds=1, j=_NPROC_PLUS_2, arch='arm', bits=[32], os='linux')),
+    ('arm64-linux-worker-1', WorkerConfig(max_builds=1, j=_NPROC_PLUS_2, arch='arm', bits=[64], os='linux')),
+    ('arm64-linux-worker-2', WorkerConfig(max_builds=1, j=_NPROC_PLUS_2, arch='arm', bits=[64], os='linux')),
+    ('rpi4-linux-worker-1', WorkerConfig(max_builds=1, j=_NPROC_PLUS_2, arch='arm', bits=[32], os='linux')),
+    ('win-worker-1', WorkerConfig(max_builds=1, j=_NPROC_PLUS_2, arch='x86', bits=[32, 64], os='windows')),
+    ('win-worker-2', WorkerConfig(max_builds=1, j=_NPROC_PLUS_2, arch='x86', bits=[32, 64], os='windows')),
```

Comment:
What does this do on Windows?

Reply:
We run in git-bash on Windows, so I *hope* it will do the right thing.

Reply:
I guess we'll see!

Reply:
ack! macOS doesn't have the `nproc` command. I tried in both zsh and bash.